### PR TITLE
增加系统合约的测试用例

### PR DESCRIPTION
**Pull request by zsq978663747**

### CODE REVIEW

**Comment by threadshare** [comment]
把你的私钥隐藏一下， 暴露了

File: config.json [hunk 1]
```diff
@@ -2,9 +2,9 @@
   "common_params": {
     "nodeos_url": "http://127.0.0.1:8888",
     "wallet_url": "http://127.0.0.1:9800",
-    "exist_account": "eosttmonster",
-    "exist_acct_active_pub": "EOS8TfBSfwjPeKCgUfCCEfJPx1UhP2CS1Xh8xoYoELtkQJQ9g99vP",
-    "exist_acct_active_pri": "5KdJZv62ZKzxhasNSojk7gC712pF9sRad4KhVyugg442qYvhWYw",
+    "exist_account": "zhangshiqi11",
+    "exist_acct_active_pub": "EOS8HrnMdHbgrchNFtV9ySE4uwCSnoix4f3JZnJcw9ryyydg2DSup",
+    "exist_acct_active_pri": "5HzG8hQfLQv4HnoSZFE4SzYCzbHV7QxJWaMzgBSgNggDDff3Lb2",
```
Comment (winlin) [inline config.json]: yourself configures should NOT be submited into the repository. just keep in local, except you want to add some common configure item.

File: eosapi/httpapi/client.py [hunk 2]
```diff
@@ -266,6 +266,7 @@ class WalletClient(HttpClient):
     def __init__(self, host='localhost', port=8888, **kwargs):
         hostname = host.split('//')[-1].split(':')[0]
         if hostname not in ['localhost', '127.0.0.1']:
+            print ('hostname',hostname)
```
Comment (winlin) [inline eosapi/httpapi/client.py]: unnecessary

File: eosapi/httpapi/http_client.py [hunk 3]
```diff
@@ -73,7 +73,7 @@ def __init__(self, nodes, **kwargs):
         '''
 
         self.nodes = cycle(self._nodes(nodes))
-        self.node_url = ''
+        self.node_url = 'http://127.0.0.1:8888'
```
Comment (winlin) [inline eosapi/httpapi/http_client.py:76]: unnecessary

File: startmonster.py [hunk 4]
```diff
@@ -64,7 +64,8 @@ def run_testcase(case_dict, common_params):
 
         cmdline = case_dict['cmdline'] + " " + params_file
         logger.info('Going to execute cmdline: {} {}'.format(case_dict['casename'], cmdline))
-        pmsg = subprocess.check_output(cmdline, stderr=subprocess.STDOUT, shell=True)
+        pmsg = subprocess.check_output(cmdline,stderr=subprocess.STDOUT, shell=True)
+        #pmsg = subprocess.check_output(cmdline, shell=True)
```
Comment (winlin) [inline startmonster.py]: unuse code should be deleted

File: testcases/01_get_request.py [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+    #判断是否与预期相等
+    if (float(account1_balance1[:-4])- float(account1_balance2[:-4])) == (float(account2_balance2[:-4]) - float(account2_balance1[:-4])):
+        print(f'{cmdline2} ===========================  ok')
+    else:
+        print(f'{cmdline2} ===========================  fail')
+
+#system contract
+def system_contract():
+    #reprod
+    cmdline = "cleos system regproducer zhangshiqi11 EOS5NKY5vhyqWNgKeNbiUx2iC6cEnooiQsYQMjCmFFMoVLUExGHba https://www.eos.store 900"
```
Comment (winlin) [inline testcases/01_get_request.py]: all the variables should be passed by config.json testcases->params, do not hardcode.

File: config.json (hunk 1 above)
Comment (zsq978663747) [reply config.json]: 好的

File: testcases/01_get_request.py (hunk 5 above)
Comment (zsq978663747) [reply testcases/01_get_request.py]: 我在考虑账户需要生成一个随机账户么，每次都创建一个随机账户，但是如果创建的话，有些浪费资源

File: README_CN.md [hunk 6]
```diff
@@ -48,3 +48,31 @@ startmonster.py 会循环调用，并把 common_params 和 testcase指定的para
 3. 运行startmoster.py
 
 ![image](./image/startmonster.png)
+
+1.注意
+
+在运行前需要打开eosapi/httpapi/http_client.py,在```class HttpClient(object):```中初始化的时候设置指定节点的url，
+例如：self.node_url = 'http://127.0.0.1:8888'
```
Comment (winlin) [inline README_CN.md]: please add the node_url as a paramater into the HttpClient() init function. We should NOT change the basic function for erverytime.

File: config.json [hunk 7]
```diff
@@ -18,6 +18,17 @@
         "demo_param1": "demo_param1",
         "demo_param2": 100
       }
+    },
+    {
+      "casename": "Testcase 01_get_request",
+      "pre_call": "echo 'This is called BEFORE cmdline call'",
+      "post_call": "echo 'This is called AFTER cmdline call'",
+      "stoponfail": "true",
+      "cmdline": "python3.6 testcases/01_get_request.py <account1> <account2> <pubkey>",
```
Comment (winlin) [inline config.json]: please use the `exist_account` parameters in the test script file, avoid add redundant account parameters

File: testcases/01_get_request.py [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+
+#get info 
+def get_info():
+    cmdline = "cleos get info"
+    process = Popen(cmdline, stdout=PIPE, stderr=PIPE,shell=True)
+    stdout, stderr = process.communicate()
+    if stdout:
+        param = json.loads(stdout)
+        print('param:',param)
+        if param['chain_id'] == '5fff1dae8dc8e2fc4d5b23b2c7665c97f9e9d8edf2b6485a86ba311c25639191':
```
Comment (winlin) [inline testcases/01_get_request.py:40]: not hardcode, please add 5fxxx into config.json

File: testcases/01_get_request.py [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+    account2_balance1 = 0
+    account2_balance2 = 0
+    #创建一个账户
+    newaccount = bpname
+    #newaccount = account_random()
+    #cmdline = f'cleos system newaccount --stake-net \"5 EOS\" --stake-cpu \"5 EOS\" --buy-ram \"2 EOS\"  {creator} {newaccount}   '
+    #process = Popen(cmdline, stdout=PIPE, stderr=PIPE,shell=True)
+    #time.sleep(1)
+    #cmdlinel = f"cleos get account {newaccount}"
+    #process = Popen(cmdlinel, stdout=PIPE, stderr=PIPE,shell=True)
```
Comment (winlin) [inline testcases/01_get_request.py:64]: remove unuse code, and keep clean

**Pull request merged**